src/common/helpers.rs

```diff
@@ -1,8 +1,16 @@
 use std::sync::atomic::{AtomicUsize, Ordering};
 
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
 use tokio::runtime;
 use tokio::runtime::Runtime;
 
+#[derive(Deserialize, Serialize, JsonSchema)]
+pub struct WriteLockStatus {
+    pub locked: bool,
+    pub error_message: Option<String>,
```

Comment:
nitpick: this object mixes both locking and unlocking state together. 

What about having two REST APIs:
- lock with optional message
- unlock (no message required)

Reply:
Sounds good. Do you like such API:
```
PUT /wtite_lock -> lock writing with optional message
PUT /write_unlock -> unlock writing
GET /write_lock -> check if locked
GET /write_unlock -> check if unlocked
```

Reply:
Good for me!

Maybe you could ask for some feedback from the cloud team if they are the main users for it.

Reply:
I implemented it. I will fix API if the cloud team will wish something else.

Reply:
```
GET /write_lock -> check if locked
GET /write_unlock -> check if unlocked
```
How is this supposed to work? Look redundant

Reply:
also `lock` is supposed to be a noun, not verb. So I would say that `PUT /write_unlock` is redundant too

Reply:
check if locked and check if unlocked. @agourlay sold me ideas about `PUT write_lock` and `PUT write_unlock` because lock and unlock have different parameters. And `GET` function imho is expected in this both paths

Reply:
What about going away from this idea of exposing `write lock` which is a bit confusing in terms of naming.

For instance, I find having a `read-only` mode clearer.

```
GET /read-only
PUT /read-only/enable 
PUT /read-only/disable
```

Reply:
Thanks for the option, LGTM. Do you accept such api @generall?

Reply:
Final api
```
POST /locks
{
  error_message: "Ram high watermark reached"
  write: true
}
```
GET is the same